srtcore/core.cpp

```diff
@@ -622,10 +623,18 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         m_CryptoSecret.typ = HAICRYPT_SECTYP_PASSPHRASE;
         m_CryptoSecret.len = (optlen <= (int)sizeof(m_CryptoSecret.str) ? optlen : (int)sizeof(m_CryptoSecret.str));
         memcpy(m_CryptoSecret.str, optval, m_CryptoSecret.len);
+#else
+        if (optlen == 0)
+            break;
+
+        LOGC(mglog.Error, log << "SRTO_PASSPHRASE: encryption not enabled at compile time");
+        throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+#endif
```

Comment:
Might be worth throwing exception when connected with the encryption disabled for consistency.
```
        if (m_bConnected) 
            throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0); 
```

Reply:
There's a rule that an erroneous situation should be reported as early as possible. This should not be done only in case when by any reason the system isn't certain of an error, or it can somehow be still avoided. In this case the connected state will not be even achieved when the password is tried to be set.

Reply:
I mean, that in no-encryption version you now don't throw exception if `optlen == 0`. While version with encryption will throw an exception if `optlen == 0` and `m_bConnected == true`.
So to make the behavior the same, I would suggest preserving the following lines for both builds:
```
        if (m_bConnected) 
            throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0); 
```

Reply:
Ah, got your point. Right, of course.